library/ssl_tls.c

```diff
@@ -1915,72 +1918,101 @@ mbedtls_ssl_mode_t mbedtls_ssl_get_mode_from_ciphersuite(
  *
  */
 #if defined(MBEDTLS_SSL_SESSION_TICKETS)
-static size_t ssl_tls13_session_save( const mbedtls_ssl_session *session,
-                                      unsigned char *buf,
-                                      size_t buf_len )
-{
-    unsigned char *p = buf;
-    size_t needed =   1                             /* endpoint */
-                    + 2                             /* ciphersuite */
-                    + 4                             /* ticket_age_add */
-                    + 2                             /* resumption_key length */
-                    + session->resumption_key_len;  /* resumption_key */
+static size_t ssl_tls13_session_serialized_size(
+                                const mbedtls_ssl_session *session )
+{
+    size_t needed = 0;
+    /* endpoint                                 1 byte
+     * ciphersuite                              2 bytes
+     * ticket_age_add                           4 bytes
+     * ticket_flags                             1 byte
+     * resumption_key length                    1 byte
+     * resumption_key          resumption_key_len bytes
+     */
+    needed += 9 + session->resumption_key_len;
 
-#if defined(MBEDTLS_HAVE_TIME)
-    needed += 8; /* start_time or ticket_received */
+#if defined(MBEDTLS_HAVE_TIME) && defined(MBEDTLS_SSL_SRV_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_SERVER )
+    {
+        needed += 8;
+    }
 #endif
 
 #if defined(MBEDTLS_SSL_CLI_C)
     if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
     {
-        needed +=   4                       /* ticket_lifetime */
-                  + 2                       /* ticket_len */
-                  + session->ticket_len;    /* ticket */
+#if defined(MBEDTLS_HAVE_TIME)
+        needed += 8;
+#endif
```

Comment:
```suggestion
#endif /* MBEDTLS_HAVE_TIME */
```

Reply:
From https://github.com/Mbed-TLS/mbedtls/pull/6069#discussion_r925794355 , it is dedicate removed.